src/test/java/org/llorllale/cactoos/matchers/HasValuesTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) for portions of project cactoos-matchers are held by
+ * Yegor Bugayenko, 2017-2018, as part of project cactoos.
+ * All other copyright for project cactoos-matchers are held by
+ * George Aristy, 2018.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.llorllale.cactoos.matchers;
+
+import java.util.Objects;
+import org.cactoos.iterable.IterableOf;
+import org.cactoos.list.ListOf;
+import org.hamcrest.Description;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.StringDescription;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+
+/**
+ * Test case for{@link HasValues}.
+ *
+ * @since 1.0.0
+ * @checkstyle MagicNumberCheck (500 lines)
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MultilineJavadocTagsCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class HasValuesTest {
+
+    @Test
+    public void contains() {
+        MatcherAssert.assertThat(
+            new HasValues<>("a", "b").matchesSafely(
+                new ListOf<>("a", "b", "c"),
+                new StringDescription()
+            ),
+            new IsEqual<>(true)
+        );
+    }
+
+    /**
+     * Verify the matcher message in case when test failed.
+     * @todo #/DEV:30m Matcher to test {@link org.hamcrest.Matcher} objects.
```

Comment:
@dgroup same here, can you put `#32`?

Reply:
Ok, will change, but I think that it will work.
From my perspective, that's different tasks and should be split into 2 separate tickets as per PDD methodology.  How dev(s) usually find that this task number should be `32`? Guess? I think it's wrong. The pattern `#/DEV:30m` gives us a new task with a new number without any references to task where it was generated. Even it 0pdd fail, we can easily do the rollback.

Reply:
@dgroup including the task's number let's `0pdd` know the issue where the puzzle was introduced